### PR TITLE
[feature] Two-factor refactor

### PR DESCRIPTION
**Pull request by sloria**
This includes @HarryRybacki 's https://github.com/CenterForOpenScience/osf.io/pull/1728, updated with the latest develop.

Sending this PR for review purposes.

---------------

## Purpose
Currently, two factor authentication is directly coupled to the log in procedure. As a result, the only way for users to verify two factor codes is to submit them at the same time as their username/password. This pull request directly addresses this inflexibility by separating the regular log in procedure from the two factor code verification procedure. 

## Changes
Users who have both two factor enabled and activated on their account will be forwarded to a 'two factor verification' page before being redirected to either their dashboard or the URI their were originally attempting to access. Additionally, the two factor code input on the /login/ page has been removed.

## Side Effects
As only a very small fraction of our users currently have two factor enabled/activated any usage bugs missed will likely go unseen for awhile.

### CODE REVIEW

File: website/settings/local [hunk 1]
```diff
@@ -0,0 +1 @@
+SEARCH_ENGINE = 'elastic'
```
Comment (HarryRybacki) [inline website/settings/local]: Was this a my bad? @sloria

**Comment by HarryRybacki** [comment]
Aside from my 1 question for @sloria all looks fine to me.

File: website/settings/local (hunk 1 above)
Comment (sloria) [inline website/settings/local]: This file can be removed.

Comment (HarryRybacki) [inline website/settings/local]: @sloria I don't actually have push rights so you'd need to pull it / merge from me again unless you want to just remove the line yourself.

**Pull request closed without merging**